docs: Update release process documentation (#1776)
kyma-project/telemetry-manager: docs/contributor/releasing.md

## Release Process

> This release process covers the steps to release new major and minor versions for the Kyma Telemetry module. At the same time, prepare a new release of the [opentelemetry-collector-components](https://github.com/kyma-project/opentelemetry-collector-components). This new release will be needed later in the release process of the telemetry-manager. The version of opentelemetry-collector-components will include the telemetry-manager version as part of its version (`{CURRENT_OCC_VERSION}-{TELEMETRY_MANAGER_VERSION}`).

Reviewer: ```suggestion
This release process covers the steps to release new major and minor versions for the Kyma Telemetry module. 

Together with the module release, prepare a new release of the [opentelemetry-collector-components](https://github.com/kyma-project/opentelemetry-collector-components). You will need this later in the release process of the Telemetry Manager. The version of `opentelemetry-collector-components` will include the Telemetry Manager version as part of its version (`{CURRENT_OCC_VERSION}-{TELEMETRY_MANAGER_VERSION}`).
```
